src/client/components/split-tile/split-tiles-row.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import { Clicker } from "../../../common/models/clicker/clicker";
+import { Colors } from "../../../common/models/colors/colors";
+import { Dimension } from "../../../common/models/dimension/dimension";
+import { DragPosition } from "../../../common/models/drag-position/drag-position";
+import { Essence, VisStrategy } from "../../../common/models/essence/essence";
+import { Split } from "../../../common/models/split/split";
+import { Stage } from "../../../common/models/stage/stage";
+import { CORE_ITEM_GAP, CORE_ITEM_WIDTH, STRINGS } from "../../config/constants";
+import { getXFromEvent, setDragData, setDragGhost } from "../../utils/dom/dom";
+import { DragManager } from "../../utils/drag-manager/drag-manager";
+import { getMaxItems } from "../../utils/pill-tile/pill-tile";
+import { DragIndicator } from "../drag-indicator/drag-indicator";
+import { AddSplit } from "./add-split";
+import "./split-tile.scss";
+import { SplitTiles } from "./split-tiles";
+
+interface SplitTilesRowProps {
+  clicker: Clicker;
+  essence: Essence;
+  menuStage: Stage;
+}
+
+interface SplitTilesRowState {
+  dragPosition?: DragPosition;
+  openedSplit?: Split;
+  overflowOpen?: boolean;
+}
+
+export class SplitTilesRow extends React.Component<SplitTilesRowProps, SplitTilesRowState> {
+  private items = React.createRef<HTMLDivElement>();
+
+  state: SplitTilesRowState = {};
+
+  private maxItems(): number {
+    const { menuStage, essence: { splits: { splits } } } = this.props;
+    return menuStage && getMaxItems(menuStage.width, splits.count());
```

Comment:
Is `menuStage` a number? `false && 1` returns `false`, that clearly isn't a number. The output type of this function looks like `Stage | number` to me.

Reply:
more like `falsy | number`

Reply:
In `SplitTilesRowProps`, `menuStage` isn't optional, so I though `Stage` may be falsy somehow. So, maybe `SplitTilesRowProps.menuStage` should be declared with `?`?

Reply:
It should.